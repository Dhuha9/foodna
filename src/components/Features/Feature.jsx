import React from 'react';
import { Row, Col, Button } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import './Features.scss';

function Feature({ index, image, title, description }) {
  const { t } = useTranslation();
  return (
    <>
      <Row className="no-gutters">
        <Col
          xs={{ span: 6, order: index % 2 === 0 ? 1 : 3 }}
          md={6}
          className="d-none d-sm-block vh-100"
          style={{ backgroundImage: `url(${image})`, backgroundSize: 'cover' }}
        ></Col>
        <Col
          xs={{ order: index % 2 === 0 ? 3 : 1 }}
          md={6}
          className="text-left justify-items-center align-items-center px-3 px-xl-5 feature-width feature_height  d-flex justify-content-center align-items-center "
        >
          <div className=" px-3 px-xl-5">
            <div className="py-5 px-3 px-lg-5 feature-text  ">
              <h1 className="my-4">{title}</h1>
              <p className="my-4">{description}</p>
              <Button className="mb-4" variant="outline-success" href="/about">
                {t(`FeatureComponent.ButtonTitle`)}
              </Button>
            </div>
            <hr className="d-block d-md-none m-0" />
          </div>
        </Col>
      </Row>
    </>
  );
}

export default Feature;

Feature.propTypes = {
  index: PropTypes.number,
  image: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
};
