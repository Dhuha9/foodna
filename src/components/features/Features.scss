.feature {
  display: flex;
  flex-direction: row;
  justify-items: center;
  align-items: center;
}

.feature-text {
  position: inherit;
  text-align: left;
  padding-left: 50px;
  justify-items: center;
  align-items: center;
}

.feature-img {
  width: 100%;
}
