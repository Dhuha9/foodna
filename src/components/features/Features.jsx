import React from 'react';
import Feature from './Feature';
import './Features.scss';

/* import M from '../../dist/img/Media.png';
import M1 from '../../dist/img/Media (1).png';
import M2 from '../../dist/img/Media (2).png'; */

function Features() {
  return <Feature />;
}

export default Features;
