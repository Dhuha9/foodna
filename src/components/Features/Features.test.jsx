import React from 'react';
import renderer from 'react-test-renderer';
import Features from './Features';

it('Features snapshot', () => {
  const tree = renderer.create(<Features />).toJSON();
  expect(tree).toMatchSnapshot();
});
