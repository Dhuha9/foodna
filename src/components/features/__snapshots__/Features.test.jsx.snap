// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="feature row"
    >
      <div
        className="col"
      >
        <img
          alt="an image from the media"
          className="feature-img"
          src="Media.png"
        />
      </div>
      <div
        className="feature-text col"
      >
        <h1
          className="mb-4"
        >
          A feature
        </h1>
        <p
          className="my-4"
        >
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the standard of the industry dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
        </p>
        <button
          className="btn btn-outline-success"
          disabled={false}
          type="button"
        >
          Learn more
        </button>
      </div>
    </div>
  </div>
</div>
`;
